src/cursor/cur_layered.c

```diff
@@ -1205,7 +1200,17 @@ __clayered_search(WT_CURSOR *cursor)
     CURSOR_API_CALL(cursor, session, ret, search, clayered->dhandle);
     WT_ERR(__cursor_needkey(cursor));
     __cursor_novalue(cursor);
-    WT_ERR(__clayered_enter(clayered, true, false, false));
+    /*
+     * FIXME-WT-15058: When inside a read committed isolation, the file cursor code expects to
+     * release the snapshot when the count of active cursors is zero. Reset the constituent cursors
+     * to adhere to that behavior. Ideally we should not be changing the active cursors counter
+     * outside of the file cursor code.
+     */
+    if (__wt_txn_read_last_check(session)) {
```

Comment:
Would you consider keeping the prior structure of passing a reset flag, but only calling reset from __clayered_enter if this condition is met?

Reply:
Done, the code looks much cleaner now